src/cscapi/mongodb_storage.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from dataclasses import asdict
+from typing import List, Optional
+
+from dacite import from_dict
+from mongoengine import (
+    ConnectionFailure,
+    Document,
+    EmbeddedDocument,
+    Q,
+    connect,
+    fields,
+)
+
+from cscapi.storage import MachineModel, SignalModel, StorageInterface
+
+logger = logging.getLogger(__name__)
+
+
+class ContextDBModel(EmbeddedDocument):
+    value = fields.StringField()
+    key = fields.StringField()
+
+
+class DecisionDBModel(EmbeddedDocument):
+    duration = fields.StringField()
+    uuid = fields.StringField()
+    scenario = fields.StringField()
+    origin = fields.StringField()
+    scope = fields.StringField()
+    simulated = fields.BooleanField()
+    until = fields.StringField()
+    type = fields.StringField()
+    value = fields.StringField()
+
+
+class SourceDBModel(EmbeddedDocument):
+    scope = fields.StringField()
+    ip = fields.StringField()
+    latitude = fields.FloatField()
+    as_number = fields.StringField()
+    range = fields.StringField()
+    cn = fields.StringField()
+    value = fields.StringField()
+    as_name = fields.StringField()
+    longitude = fields.FloatField()
+
+
+class SignalDBModel(Document):
+    alert_id = fields.SequenceField(unique=True)
+    created_at = fields.StringField()
+    machine_id = fields.StringField(max_length=128)
+    scenario_version = fields.StringField(null=True)
+    message = fields.StringField(null=True)
+    uuid = fields.StringField()
+    start_at = fields.StringField(null=True)
+    scenario_trust = fields.StringField(null=True)
+    scenario_hash = fields.StringField(null=True)
+    scenario = fields.StringField(null=True)
+    stop_at = fields.StringField(null=True)
+    sent = fields.BooleanField(default=False)
+    context = fields.EmbeddedDocumentListField(ContextDBModel)
+    decisions = fields.EmbeddedDocumentListField(DecisionDBModel)
+    source = fields.EmbeddedDocumentField(SourceDBModel)
+
+
+class MachineDBModel(Document):
+    machine_id = fields.StringField(max_length=128, unique=True)
+    token = fields.StringField()
+    password = fields.StringField()
+    scenarios = fields.StringField()
+    is_failing = fields.BooleanField(default=False)
+
```

Comment:
For consistency with sql table name, could you please set the same table name by adding this line: 
```
meta = {"collection": "machine_models"}
```